Crash when uploading file and not enough space to copy

### Steps to reproduce
1. Upload a file that is larger than the amount of free space left on the device.
2. Answer yes to the question about moving the file to next cloud instead of copying.

### Expected behaviour
- file should upload without an error

### Actual behaviour
- program dump takes place pasted below.

************ CAUSE OF ERROR ************

java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=2, result=1, data=Intent { (has extras) }} to activity {com.nextcloud.client/com.owncloud.android.ui.activity.FileDisplayActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.endsWith(java.lang.String)' on a null object reference
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5304)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5343)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:54)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2214)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7842)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.endsWith(java.lang.String)' on a null object reference
	at com.owncloud.android.ui.activity.FileDisplayActivity.requestUploadOfFilesFromFileSystem(FileDisplayActivity.java:964)
	at com.owncloud.android.ui.activity.FileDisplayActivity.requestUploadOfFilesFromFileSystem(FileDisplayActivity.java:959)
	at com.owncloud.android.ui.activity.FileDisplayActivity.onActivityResult(FileDisplayActivity.java:889)
	at android.app.Activity.dispatchActivityResult(Activity.java:8382)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5297)
	... 13 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30170151
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: google
Device: blueline
Model: Pixel 3
Id: SPB5.210812.002
Product: blueline

************ FIRMWARE ************
SDK: 31
Release: 12
Incremental: 7671067

--- a/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java
+++ b/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java
@@ -949,7 +949,7 @@ public class FileDisplayActivity extends FileActivity
     }
 
     private void requestUploadOfFilesFromFileSystem(String localBasePath, String[] filePaths, int resultCode) {
-        if (filePaths != null) {
+        if (localBasePath != null && filePaths != null) {
             if (!localBasePath.endsWith("/")) {
                 localBasePath = localBasePath + "/";
             }

